avocado/core/nrunner.py

```diff
@@ -601,15 +604,15 @@ def _run_unittest(cls, uri, queue):
         queue.put(output)
 
     def run(self):
-        if not self.runnable.uri:
-            error_msg = 'uri is required but was not given'
+        if not self.unittest_name:
+            error_msg = 'URI is required but was not given or it is invalid.'
```

Comment:
Based on the previous comment about typing, I'd simply state that the "URI is not valid because it could not be converted to an unittest dotted name" or something like that (which covers empty strings).